File: Java/RecyclerViewJava/app/src/main/java/com/tmdhoon/recyclerviewjava/MainActivity.java
package com.tmdhoon.recyclerviewjava;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import com.tmdhoon.recyclerviewjava.databinding.ActivityMainBinding;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private MainAdapter mainAdapter;
    private LinearLayoutManager linearLayoutManager;

    private List<MainData> list;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        list = new ArrayList<>();

        linearLayoutManager = new LinearLayoutManager(getApplicationContext());

        binding.recyclerview.setLayoutManager(linearLayoutManager);

        mainAdapter = new MainAdapter(list);

        binding.recyclerview.setAdapter(mainAdapter);

        binding.btAdd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                list.add(new MainData(binding.etInput.getText().toString()));
                mainAdapter.notifyDataSetChanged();
                binding.etInput.setText("");
            }
        });

    }
}

File: Kotlin/KotlinPractice/app/src/main/java/com/example/kotlinpractice/MainActivity.kt
package com.example.kotlinpractice

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import com.example.kotlinpractice.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val button : Button = findViewById(R.id.button)

        button.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {

            }
        })

        // 람다식 사용조건

        // 1. Kotlin interface가 아닌 자바 인터페이스여야 합니다.
        // 2. 그 인터페이스는 딱 하나의 메소드만 가져야 합니다.

        button.setOnClickListener {

        }
    }
}